spec/abstractops.html

```diff
@@ -1029,32 +1029,41 @@ <h1>ISO 8601 grammar</h1>
-          TimeZoneIANANameComponent
-          TimeZoneIANANameComponent `/` TimeZoneIANANameTail
+          [+Legacy] `+`
+          [+Legacy] DecimalDigit
+
+      TimeZoneIANANameComponent[Legacy] :
+          > Readability note: This production matches 1 to 14 characters.
+          TZLeadingChar but not `.`
+          TZLeadingChar TZChar[?Legacy] but not `..`
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+          TZLeadingChar TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy] TZChar[?Legacy]
+
+      TimeZoneIANANameTail[Legacy] :
+          TimeZoneIANANameComponent[?Legacy]
+          TimeZoneIANANameComponent[?Legacy] `/` TimeZoneIANANameTail[?Legacy]
 
       TimeZoneIANALegacyName :
-          `Etc/GMT0`
-          `GMT0`
-          `GMT-0`
-          `GMT+0`
-          `EST5EDT`
-          `CST6CDT`
-          `MST7MDT`
-          `PST8PDT`
+          TimeZoneIANANameTail[+Legacy] [&gt; but only if `etc/gmt` |ASCIISign| |UnpaddedHour| matches StringToCodePoints(the ASCII-lowercase of CodePointsToString(the sequence of code points matched by |TimeZoneIANANameTail|))]
+          TimeZoneIANANameTail[+Legacy] [&gt; but only if the sequence of code points matched by |TimeZoneIANANameTail| is an ASCII-case-insensitive match for an element of « *"Etc/GMT0"*, *"GMT0"*, *"GMT-0"*, *"GMT+0"*, *"EST5EDT"*, *"CST6CDT"*, *"MST7MDT"*, *"PST8PDT"* »]
```

Comment:
I think this would be cleaner if we generalized "ASCII-lowercase" to sequences of code points.
```suggestion
          TimeZoneIANANameTail[+Legacy] [&gt; but only if `etc/gmt` |ASCIISign| |UnpaddedHour| matches the ASCII-lowercase of the sequence of code points matched by |TimeZoneIANANameTail|]
          TimeZoneIANANameTail[+Legacy] [&gt; but only if the sequence of code points matched by |TimeZoneIANANameTail| is an ASCII-case-insensitive match for an element of « *"Etc/GMT0"*, *"GMT0"*, *"GMT-0"*, *"GMT+0"*, *"EST5EDT"*, *"CST6CDT"*, *"MST7MDT"*, *"PST8PDT"* »]
```

Reply:
This isn't accepted by grammarkdown for some reason, but I've gotten it to work by adding superfluous parentheses around |TimeZoneIANANameTail|.